yarpl/include/yarpl/flowable/CancelingSubscriber.h

```diff
@@ -13,16 +13,21 @@ namespace flowable {
  * A Subscriber that always cancels the subscription passed to it.
  */
 template <typename T>
-class CancelingSubscriber final : public InternalSubscriber<T> {
+class CancelingSubscriber final : public BaseSubscriber<T> {
```

Comment:
Why is CancelingSubscriber deriving from the BaseSubscriber? Do we need any of the functionality?

Reply:
It forces CancelingSubscriber to obey the semantics of "onNext will never be called after cancel()" that we've adopted for BaseSubscriber. It doesn't *need* to do this, but there's not much reason not to.